allensdk/brain_observatory/behavior/data_objects/stimuli/presentations.py

```diff
@@ -0,0 +1,152 @@
+                                               possible_names)
+        stimulus_names = stimulus_table[stimulus_name_column].unique()
+
+        for stim_name in sorted(stimulus_names):
+            specific_stimulus_table = stimulus_table[stimulus_table[
+                                                         stimulus_name_column] == stim_name]  # noqa: E501
+            # Drop columns where all values in column are NaN
+            cleaned_table = specific_stimulus_table.dropna(axis=1, how='all')
+            # For columns with mixed strings and NaNs, fill NaNs with 'N/A'
+            for colname, series in cleaned_table.items():
+                types = set(series.map(type))
+                if len(types) > 1 and str in types:
+                    series.fillna('N/A', inplace=True)
+                    cleaned_table[colname] = series.transform(str)
+
+            interval_description = (f"Presentation times and stimuli details "
+                                    f"for '{stim_name}' stimuli. "
+                                    f"\n"
+                                    f"Note: image_name references "
+                                    f"control_description in "
+                                    f"stimulus/templates")
+            presentation_interval = create_stimulus_presentation_time_interval(
+                name=f"{stim_name}_presentations",
+                description=interval_description,
+                columns_to_add=cleaned_table.columns
+            )
+
+            for row in cleaned_table.itertuples(index=False):
+                row = row._asdict()
+
+                presentation_interval.add_interval(
+                    **row, tags='stimulus_time_interval', timeseries=ts)
+
+            nwbfile.add_time_intervals(presentation_interval)
+
+        return nwbfile
+
+    @classmethod
+    def from_nwb(cls, nwbfile: NWBFile) -> "Presentations":
+        # Note: using NwbApi class because ecephys uses this method
+        # TODO figure out how behavior and ecephys can share this method
+        nwbapi = NwbApi.from_nwbfile(nwbfile=nwbfile)
+        df = nwbapi.get_stimulus_presentations()
+
+        df['is_change'] = is_change_event(stimulus_presentations=df)
+        return Presentations(presentations=df)
+
+    @classmethod
+    def from_stimulus_file(
+            cls, stimulus_file: StimulusFile,
+            limit_to_images: Optional[List] = None) -> "Presentations":
+        """Get stimulus presentation data.
+
+        NOTE: Uses timestamps that do not account for monitor delay.
+
+        :param limit_to_images
+            Only return images given by these image names
+
+        :returns: pd.DataFrame --
+            Table whose rows are stimulus presentations
+            (i.e. a given image, for a given duration, typically 250 ms)
+            and whose columns are presentation characteristics.
+        """
+        stimulus_timestamps = StimulusTimestamps.from_stimulus_file(
```

Comment:
Same issue as previous comments, sync should take precendence over stimulus *.pkl when it exists.